simulations/src/network/mod.rs

```diff
@@ -106,20 +106,24 @@ pub struct Network<M> {
     network_time: NetworkTime,
     messages: Vec<(NetworkTime, NetworkMessage<M>)>,
     from_node_receivers: HashMap<NodeId, Receiver<NetworkMessage<M>>>,
+    from_node_broadcast_receivers: HashMap<NodeId, Receiver<NetworkMessage<M>>>,
```

Comment:
Is this really necessary?  A broadcasting can be just treated as sending a message to each node in the network.  There should be no duplication of handlers.

Reply:
Network interface communicates with the Network via the channels. To implement broadcasting two options comes to mind: Use enum to distinguish adhoc and broadcast message (this was done in a first implementation), or use dedicated channel. I feel that later solution plays nice with how network collects messages after each step and dispatches it (taking message delays into account).